app/js/lib/intent-parser/refiners/reminder-refiner.js

```diff
@@ -0,0 +1,28 @@
+import ReminderConfirmation from '../reminder-confirmation';
+
+const p = Object.freeze({
+  confirmation: Symbol('confirmation'),
+});
+
+export default class ReminderRefiner {
+  constructor() {
+    this[p.confirmation] = new ReminderConfirmation();
+  }
+
+  refine(obj = {}) {
+    if (obj.cleaned.toLowerCase().startsWith('remind')
+      && obj.time !== null
+      && obj.time.length === 1
+      && obj.time[0].start !== null
+      && obj.time[0].end === null
+      && obj.recipients !== null
+      && obj.recipients.length > 0
+      && obj.action !== null) {
```

Comment:
Maybe assign some of the combinations to variables for readability, would help understanding too - these variable names may not be appropriate.

```JS
const hasRecipients = obj.recipients !== null && obj.recipients.length > 0;
const hasTime = obj.time !== null && obj.time.length === 1;
// etc...

if (hasTime && hasRecipients /* etc... */ ) {}
```